### PR TITLE
Dismissable alerts

### PR DESCRIPTION
**Pull request by tcbegley**
This PR adds a `dismissable` prop to `Alert` which, when set to `True`, adds a dismiss button to the alert. It can be used alongside the existing `is_open` prop to display or dismiss alerts. See the following screenshot:

image

I have also taken the opportunity to improve the alert documentation, and add some more detailed examples.

### CODE REVIEW

File: docs/components_page/components/alert/__init__.py [hunk 1]
```diff
@@ -1,19 +1,79 @@
...
+        ),
+        ExampleContainer(alert_content),
+        HighlightedSource(alert_content_source),
+        html.H4("Dismissing"),
+        html.P(
+            dcc.Markdown(
+                "Set `dismissable=True` to add a dismiss button to the alert "
+                "which closes the alert on click. You can also use the "
+                "`is_open` property in callbacks to show or hide the alert. "
+                "By default the alert appeaers and disappears with a fade "
```
Comment (pbugnion) [inline docs/components_page/components/alert/__init__.py]: appears

File: docs/components_page/components/alert/__init__.py [hunk 2]
```diff
@@ -1,19 +1,79 @@
...
+                "animation, to disable this simply set `fade=False`."
```
Comment (pbugnion) [inline docs/components_page/components/alert/__init__.py]: I feel like that sentence could benefit from being split into two:

> By default the alert appeaers and disappears with a fade animation. To disable this simply set fade=False.

**Comment by pbugnion** [comment]
For reference, this is what the updated documentation looks like:

screenshot

**Comment by pbugnion** [comment]
Looks great!

**Comment by jchang10** [comment]
I just noticed that 'dismissable' only works if you give the alert an id. Seems like an unnecessary burden to have to give an id to an Alert that will never get used on the server. Only needed to hide it on the client, right?

Just for future reference, I wonder, is this a quirk about Dash not being able to propagate events to id-less components?

**Comment by tcbegley** [comment]
Hey @jchang10,

Thanks a lot for raising this. I haven't been able to reproduce it though. This is the test app I ran:

```python
import dash
import dash_bootstrap_components as dbc

app = dash.Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])

app.layout = dbc.Alert("This is an alert", dismissable=True)

if __name__ == "__main__":
    app.run_server(debug=True, port=8888)
```

The Alert dismissed as I would have expected when clicking on the dismiss button. These are the versions of Dash libraries I was using:

```
dash==0.41.0
dash-bootstrap-components==0.4.0
dash-core-components==0.46.0
dash-html-components==0.15.0
dash-renderer==0.22.0
dash-table==3.6.0
```

Could you share an example where you see the behaviour you describe?

**Comment by jchang10** [comment]
Yes, exact same versions, but I am still seeing the same problem. 

The only difference with my test is this line. Change:
    app.layout = dbc.Alert("This is an alert", dismissable=True)
to:
    import dash_html_components as html
app.layout = html.Div( dbc.Alert("This is an alert", dismissable=True), id='testdiv')

IE, add a Div wrapper or even a dbc.Card, etc. If I add an id to Alert, it works, tho.

**Comment by tcbegley** [comment]
@jchang10 I managed to reproduce it, thanks. Also, that is very weird! I'll try to figure out what the problem is.

I'm going to make an issue for this for visibility.

**Pull request merged**